src/lib/bootstrap_controller/bootstrap_controller.ml

```diff
@@ -273,6 +306,41 @@ end = struct
       let root_sync_ledger =
         Root_sync_ledger.create ledger_db ~logger:t.logger ~trust_system
       in
+      (let%bind () = Ivar.read ask_best_tip_signal in
+       Logger.info logger
+         "Requesting peers for their best tip to eargly start bootstrap"
+         ~module_:__MODULE__ ~location:__LOC__ ;
+       match%bind
+         download_best_tip ~network ~logger ~trust_system ~verifier
+           initial_root_transition
+       with
+       | Ok
+           ( ((peer_best_tip, _) as peer_best_tip_with_validation)
+           , peer_root_with_validation
+           , peer ) ->
+           let best_tip_candidate_state =
+             External_transition.consensus_state peer_best_tip.data
+           in
+           if not @@ worth_getting_root t best_tip_candidate_state then
+             Deferred.unit
```

Comment:
Shouldn't this be an error? If someone sent you a best tip that is not worth getting the root for, then something is wrong, isn't it?

Reply:
Removed this check because it is implicitly unneeded.  Currently, we download from other peers their best_tip and, in parallel, sync with gossiped transitions through `transition _reader`. Though this check, there could be a case that the downloaded `best_tip` is not worth getting because the gossip network provided a better one. With my revisions, the first thing that we will do when bootstrapping is ask other nodes for their `best_tip`s and then synchronously sync with gossiped transitions from other peers. So, the tips that we downloaded are always worth getting. As a result, this check is unneeded anymore.